server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/Saml2BearerGrantAuthenticationConverter.java

```diff
@@ -0,0 +1,347 @@
+     * {@link Authentication} token
+     *
+     * @return the default assertion validator strategy
+     */
+    public static Converter<OpenSaml4AuthenticationProvider.AssertionToken, Saml2ResponseValidatorResult> createDefaultAssertionValidator() {
+
+        return createDefaultAssertionValidatorWithParameters(
+                params -> params.put(SAML2AssertionValidationParameters.CLOCK_SKEW, Duration.ofMinutes(5)), true);
+    }
+
+    /**
+     * Construct a default strategy for converting a SAML 2.0 Assertion and
+     * {@link Authentication} token into a {@link Saml2Authentication}
+     *
+     * @return the default response authentication converter strategy
+     */
+    static Converter<OpenSaml4AuthenticationProvider.AssertionToken, AbstractAuthenticationToken> createDefaultAssertionAuthenticationConverter() {
+        return assertionToken -> {
+            Assertion assertion = assertionToken.getAssertion();
+            Saml2AuthenticationToken token = assertionToken.getToken();
+            String username = assertion.getSubject().getNameID().getValue();
+            Map<String, List<Object>> attributes = OpenSaml4AuthenticationProvider.getAssertionAttributes(assertion);
+            List<String> sessionIndexes = OpenSaml4AuthenticationProvider.getSessionIndexes(assertion);
+            DefaultSaml2AuthenticatedPrincipal principal = new DefaultSaml2AuthenticatedPrincipal(username, attributes,
+                    sessionIndexes);
+            String registrationId = token.getRelyingPartyRegistration().getRegistrationId();
+            principal.setRelyingPartyRegistrationId(registrationId);
+            return new Saml2Authentication(principal, token.getSaml2Response(),
+                    AuthorityUtils.createAuthorityList("ROLE_USER"));
+        };
+    }
+
+    /**
+     * Construct a default strategy for validating each SAML 2.0 Assertion and associated
+     * {@link Authentication} token
+     *
+     * @param validationContextParameters a consumer for editing the values passed to the
+     *                                    {@link ValidationContext} for each assertion being validated
+     * @return the default assertion validator strategy
+     * @since 5.8
+     */
+
+    @Override
+    public Authentication convert(HttpServletRequest request) throws AuthenticationException {
+        RelyingPartyRegistration relyingPartyRegistration = relyingPartyRegistrationResolver.resolve(request, null);
+
+        String serializedAssertion = request.getParameter("assertion");
+        byte[] decodedAssertion = Saml2Utils.samlDecode(serializedAssertion);
+        String assertionXml = new String(decodedAssertion, StandardCharsets.UTF_8);
+
+        Assertion assertion = parseAssertion(assertionXml);
+        Saml2AuthenticationToken authenticationToken = new Saml2AuthenticationToken(relyingPartyRegistration, assertionXml);
+        process(authenticationToken, assertion);
```

Comment:
here we validate signature, but we are not confident that we have the correct IdP

Did not found a security problem, but the error messages are misleading because then you see always signature error message